File: Problems/Mathematics/Day-01/sol/Saurav_Gitte/Solution1.cpp
/*
Problem Statement:
There are n people standing in a circle (n is even). Each person looks at the person
standing exactly opposite them. Given that person a looks at person b, determine
the person that person c is looking at.

---------------------------------------
Approach:
If 'a' looks at 'b', then the distance between a and b must be exactly n/2.
Let dist = |a - b|. Then n = 2 * dist.
Once n is known, the person opposite to c is:
- c + dist, if it does not exceed n
- otherwise, c - dist

If c is greater than n, then the configuration is invalid.

----------------------------------------

Time Complexity: O(1)

Space Complexity: O(1)

----------------------------------------
Example:
Input:
a = 1, b = 3, c = 2

Process : 
1. Calculate dist = |1 - 3| = 2
2. Calculate n = 2 * dist = 4  
3. Since c = 2, calculate opposite:
   - c + dist = 2 + 2 = 4 (which is <= n)
   - So, the person opposite to c is 4.

Output:
4

--------------------------------
*/

#include <bits/stdc++.h>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int a, b, c;
        cin >> a >> b >> c;


        int d = abs(a - b);
        int n = 2 * d;
        // 2*10^8 fits in int.

        // Invalid configuration, Answer does not exist
        if (d == 0 || c > n) {
            cout << -1 << "\n";
            continue;
        }

        int ans;
        if (c + d <= n)
            ans = c + d;
        else
            ans = c - d;

        cout << ans << "\n";
    }
    return 0;
}

File: Problems/Mathematics/Day-01/sol/Prashant_dwivedi/Solution1.java
// 1. Problem Statement -------------------------------------------------
/*
There are n people standing in a circle, where n is even.
Each person looks at the person standing exactly opposite to them.

You are given three distinct integers a, b, and c.
It is known that person a is looking at person b.

Determine the person that person c is looking at.
If such a person cannot exist, print -1.
*/

// 2. Approach -----------------------------------------------------------
/*
The pair (a, b) fixes the entire circle.
Once the circle is fixed, every person has exactly one opposite person.

If person c does not fit into this fixed circle, the answer is -1.
Otherwise, move half the circle from c to find the person they are looking at.
*/

// 3. Complexity ---------------------------------------------------------
/*
Time Complexity: O(1) per test case
Space Complexity: O(1)
*/

//4. problem submission link-----------------------------------
// https://codeforces.com/problemset/status?my=on

// Compiling Code in Java (Solution1.java) -------------------------------

import java.util.*;

public class Solution1 {

    public static int findOpposite(int a, int b, int c) {
        int n = 2 * Math.abs(a - b);
        if (a > n || b > n || c > n) {
            return -1;
        }
        int half = n / 2;
        if (c + half <= n) {
            return c + half;
        }
        return c - half;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            int a = sc.nextInt();
            int b = sc.nextInt();
            int c = sc.nextInt();
            System.out.println(findOpposite(a, b, c));
        }
        sc.close();
    }
}
